src/engraving/dom/cmd.cpp

```diff
@@ -3375,9 +3375,16 @@ void Score::cmdExplode()
     // reset selection to top staff only
     // force complete measures
     Segment* startSegment = selection().startSegment();
-    Segment* endSegment = selection().endSegment();
+    Segment* endSegment   = selection().endSegment();
     Measure* startMeasure = startSegment->measure();
-    Measure* endMeasure = endSegment ? endSegment->measure() : lastMeasure();
+    Measure* endMeasure   = nullptr;
+    if (!endSegment) {
+        endMeasure = lastMeasure();
+    } else if (endSegment == tick2segment(endSegment->measure()->tick())) {
```

Comment:
Perhaps the call to `tick2segment` can be avoided. Maybe `endSegment->tick() == endSegment->measure()->tick()`? Or `!endSegment->prev()`? Or `endSegment == endSegment->measure()->first()`?
(I believe `tick2segment` is somewhat expensive)